File: pset2/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])

{
    // check command line argument
    if (argc != 2)
    {

        printf("Hello, Please enter a valid command line argument\n");
        return 1;
    }
    // plain text string from user
    printf("plaintext: ");
    string s = get_string();

    // convert argv to integer
    int key = atoi(argv[1]);
    printf("ciphertext: ");

    // iterate thru string and output ciphertext
    for (int i = 0; i < strlen(s); i++)
    {

        int cipherkey = key % 26;

        // check if character is a letter
        if (s[i] < 65 || (s[i] > 90 && s[i] < 97) || s[i] > 122)

        {
            printf("%c", s[i]);
        }

        // if character is a letter and does not need to wrap around
        else if (toupper(s[i]) + cipherkey <= 90)
        {
            printf("%c", s[i] + cipherkey);
        }
        else
        {
            // if character is a letter and does need to wrap around
            printf("%c", s[i] + cipherkey - 26);
        }
    }
    printf("\n");
    return 0;
}

File: pset6/caesar.py
from cs50 import get_string
from cs50 import sys


def main():

    # check command line argument
    if len(sys.argv) != 2:
        print("Hello, Please enter a valid command line argument")
        exit(1)

    key = int(sys.argv[1])

    # plain text string from user
    plaintext = get_string("Plaintext: ")
    print("ciphertext: ", end="")

    # iterate thru string and output ciphertext
    for i in range(len(plaintext)):
        # check for uppercase
        if str.isupper(plaintext[i]):
            upper = (((ord(plaintext[i]) - 65) + key) % 26) + 65
            print(chr(upper), end="")

        # check for lowercase
        elif str.islower(plaintext[i]):
            lower = (((ord(plaintext[i]) - 97) + key) % 26) + 97
            print(chr(lower), end="")
        # else print
        else:
            print("{}".format(plaintext[i]), end="")

    # print new line
    print()


if __name__ == "__main__":
    main()
